contracts/registries/LicenseRegistry.sol

```diff
@@ -438,22 +440,52 @@ contract LicenseRegistry is ILicenseRegistry, AccessManagedUpgradeable, UUPSUpgr
         }
     }
 
+    /// @dev Calculate the earliest expiration time of the child IP with both parent IPs and license terms
+    /// @param earliestParentIpExp The earliest expiration time of among all parent IPs
+    /// @param licenseTemplate The address of the license template where the license terms are created
+    /// @param licenseTermsIds The license terms the child IP is registered with
+    function _calculateEarliestExpireTime(
+        uint256 earliestParentIpExp,
+        address licenseTemplate,
+        uint256[] calldata licenseTermsIds
+    ) internal view returns (uint256 earliestExp) {
+        earliestExp = earliestParentIpExp;
+        uint256 licenseExp = ILicenseTemplate(licenseTemplate).getEarlierExpireTime(licenseTermsIds, block.timestamp);
+        if (licenseExp > 0 && (licenseExp < earliestExp || earliestExp == 0)) earliestExp = licenseExp;
```

Comment:
I see some code duplication here with line 225-226 on contracts/registries/LicenseRegistry.sol and line 248 on contracts/modules/licensing/PILicenseTemplate.sol, do you plan to create a helper function to avoid some code duplication?

Reply:
Yes, extracted a common helper function with commit [94eb0cb](https://github.com/storyprotocol/protocol-core-v1/pull/129/commits/94eb0cbb540eb9a24f070dc31377d0ceaa090166).